lib/compilers/ldc.ts

```diff
@@ -56,42 +59,43 @@ export class LDCCompiler extends BaseCompiler {
         return super.runExecutable(executable, executeParameters, homeDir);
     }
 
-    getSharedLibraryPathsAsArguments() {
+    override getSharedLibraryPathsAsArguments() {
         return [];
     }
 
-    optionsForFilter(filters, outputFilename) {
+    override optionsForFilter(filters, outputFilename) {
         let options = ['-gline-tables-only', '-of', this.filename(outputFilename)];
         if (filters.intel && !filters.binary) options = options.concat('-x86-asm-syntax=intel');
         if (!filters.binary) options = options.concat('-output-s');
         return options;
     }
 
-    getArgumentParser() {
+    override getArgumentParser() {
         return LDCParser;
     }
 
-    filterUserOptions(userOptions) {
+    override filterUserOptions(userOptions) {
         return userOptions.filter(option => option !== '-run');
     }
 
-    isCfgCompiler() {
+    override isCfgCompiler() {
         return true;
     }
 
-    couldSupportASTDump(version) {
+    override couldSupportASTDump(version) {
         const versionRegex = /\((\d\.\d+)\.\d+/;
         const versionMatch = versionRegex.exec(version);
         return versionMatch ? semverParser.compare(versionMatch[1] + '.0', '1.4.0', true) >= 0 : false;
     }
 
-    async generateAST(inputFilename, options) {
+    override async generateAST(inputFilename, options) {
         // These options make LDC produce an AST dump in a separate file `<inputFilename>.cg`.
         const newOptions = options.concat('-vcg-ast');
         const execOptions = this.getDefaultExecOptions();
+        // TODO(#4609) generateAST expects to return a ResultLine[] not a string
```

Comment:
This TODO refers to itself? Should be a separate issue unless you plan o nfixing it before merging?

Reply:
I was trying to do the same thing we did in #4490, I can make a separate issue for this though